### PR TITLE
fix: set env.ATOM_HOME to temp directory

### PR DESCRIPTION
**Pull request by aminya**
Was brought up in [this comment](https://github.com/steelbrain/package-deps/pull/335#issuecomment-813812213)

### CODE REVIEW

**Comment by UziTech** [comment]
Can you give an example of how this would help? This only sets ATOM_HOME while the tests are running. So this would only affect apm if it was run in a test.

**Comment by aminya** [comment]
> Can you give an example of how this would help? This only sets ATOM_HOME while the tests are running. So this would only affect apm if it was run in a test.

This removes the need for `testPackages` option if the package itself uses `package-deps`. It also allows properly testing the installation of dependencies.

**Comment by UziTech** [comment]
I don't know if this would actually help in that case. This sets the `ATOM_HOME` property of the `process.env` object in this process. This does not set the `ATOM_HOME` environment variable. So I don't think spawning a new `apm` process to install packages like `atom-package-deps` does would not use this variable anyways.

**Comment by aminya** [comment]
Apm that is `spawn`ed by `package-deps` uses `process.env` by default.

https://github.com/steelbrain/package-deps/blob/af7b6d062c3402270481f3f0710fe26abc470990/src/helpers/index.ts#L131

**Comment by aminya** [comment]
> Could you write a test for this? maybe something that just spawns `apm` and installs a package and checks that it is in the correct folder.

Done.

**Review by UziTech — CHANGES_REQUESTED** [review]
I was thinking more along the lines of just running `apm install atom-ide-base` and checking that it is in the correct folder. It seems like the test you wrote is testing atom-ide-base's functionality.

**Comment by aminya** [comment]
It tests the end-to-end functionality I care about. It tests package-deps which uses apm underneath. Testing apm directly isn't very helpful to me.

**Comment by UziTech** [comment]
I would rather test APM directly so a bug in atom-ide-base or atom-package-deps doesn't make these tests fail.

File: spec/runner.js [hunk 1]
```diff
@@ -4,4 +4,5 @@ import { createRunner } from "../";
 
 export default createRunner({
 	specHelper: true,
+	testPackages: ["atom-ide-base"]
```
Comment (UziTech) [inline spec/runner.js:7]: This change isn't really needed

```suggestion
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -28,7 +28,9 @@ jobs:
     - name: Install dependencies
       run: apm ci
     - name: Run tests 👩🏾‍💻
-      run: npm run test
+      run: |
+        apm install atom-ide-base
+        npm run test
```
Comment (UziTech) [inline .github/workflows/main.yml:33]: This change isn't really needed

```suggestion
      run: npm run test
```

File: spec/apm-spec.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+"use babel";
+import { spawnSync } from "child_process";
+
+describe("apm", () => {
+	it("installs the dependencies in the correct test path", function () {
+		spawnSync("apm install atom-ide-base", { shell: true, stdio: "inherit", encoding: "utf8" });
```
Comment (aminya) [inline spec/apm-spec.js:6]: Now it doesn't install

**Comment by aminya** [comment]
This is strange. The tests fail locally.

**Comment by UziTech** [comment]
> The tests fail locally.

Maybe because you already have it installed?

File: spec/apm-spec.js (hunk 3 above)
Comment (UziTech) [reply spec/apm-spec.js:6]: Maybe we should skip this test unless `process.env.CI` is set.

**Comment by aminya** [comment]
> > The tests fail locally.
> 
> Maybe because you already have it installed?

Why is that related? It shouldn't be. Package-deps tests were successful.

File: spec/apm-spec.js (hunk 3 above)
Comment (UziTech) [reply spec/apm-spec.js:6]: Actually it should install into the temp directory so it shouldn't matter if it is already installed

**Comment by UziTech** [comment]
> Package-deps tests were successful.

That is because you installed atom-ide-base and symlinked it before the tests started. If you had commented out the `apm install atom-ide-base` line from that test it still would have passed.

**Review by UziTech — APPROVED** [review]

**Pull request merged**